CustomerContactBook/Services/GroupService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using CustomerContactBook.Database;
+using CustomerContactBook.Database.Tables;
+using CustomerContactBook.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CustomerContactBook.Services
+{
+    public class GroupService
+    {
+        private readonly ContactBookContext _context;
+
+        public GroupService(ContactBookContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<CustomerGroupModel>> GetGroups()
+        {
+            var result =  await _context.Groups.ToListAsync();
+            return result.Select(toGroupModel).ToList();
+        }
+
+        public async Task<CustomerGroupModel> GetCustomerGroup(long id)
+        {
+            var customerGroup = await _context.Groups.FindAsync(id);
+            if (customerGroup == null)
+            {
+                return null;
+            }
+            return toGroupModel(customerGroup);
+        }
+
+        public async Task<bool?> PutCustomerGroup(long id, CustomerGroupModel model)
```

Comment:
Put is HTTP method and it's clear why you may use it in the controller's method names. But there is service layer and better to use regular language (update, create, ...)